### ISSUE TITLE
Move inlineCallbacks test from twisted.test.test_failure to somewhere better (test_defer?)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5955]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5955|
|Type|enhancement|
|Created|2012-09-07 12:30:59Z|
|Branch|https://github.com/twisted/twisted/tree/move-inlinecbtest-5955|

Importing `twisted.internet` in a `twisted.python` test module is not ideal. The tests involving `inlineCallbacks` in `test_failure` should probably be moved somewhere else, e.g. the relevant `test_defer`.

<details><summary>Searchable metadata</summary>

```
trac-id__5955 5955
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_move_inlinecbtest_5955 branches/move-inlinecbtest-5955
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347021059000000 1347021059000000
changetime__1364477889000000 1364477889000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [37306]) Branching to 'move-inlinecbtest-5955'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [37307]) move test, add news file. refs [#5955](https://github.com/twisted/twisted/issues/5955)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Forced a [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/move-inlinecbtest-5955).

 1. moved test to test_defer
 1. the test uses `getDivisionFailure` from `test.test_failure`. it's being imported now but it seems better to expose this elsewhere (or maybe duplicate)?
 1. moved the py3 todo statement and added notes about [#6008](https://github.com/twisted/twisted/issues/6008) being a unittest todo support ticket, found it confusing with the [#5949](https://github.com/twisted/twisted/issues/5949) right above it..

Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

  1. Rather than importing `getDivisionFailure`, you can probably just duplicate it as a local function.
  2. The test should use `failureResultOf` to extract the failure from the deferred.
  3. The assertion could be made much stronger (and match the docstring) if the entire extracted traceback was compared to the original traceback.
Please resubmit for review after addressing the above points.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [tom.prince](#note_4):
>   1. Rather than importing `getDivisionFailure`, you can probably just duplicate it as a local function.

Done.

>   2. The test should use `failureResultOf` to extract the failure from the deferred.

Fixed.

>   3. The assertion could be made much stronger (and match the docstring) if the entire extracted traceback was compared to the original traceback.

The assertion has been expanded, and while it's an improvement, i couldn't figure out how to get the original traceback to compare to, but it also feels this is more out of scope for this ticket.

Forced a new [build](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/move-inlinecbtest-5955).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

This looks good. Please merge.



>> 3. The assertion could be made much stronger (and match the docstring) if the entire extracted traceback was compared to the original traceback.
> The assertion has been expanded, and while it's an improvement, i couldn't figure out how to get the original traceback to compare to, but it also feels this is more out of scope for this ticket.
I like the new assertions. It would be nice if there was some helpers for doing this, so that the reader doesn't need know how tracebacks look.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [37859]) Merge move-inlinecbtest-5955: Move test_inlineCallbacksTracebacks from test_failure to test_defer.

Author: thijs
Reviewer: tom.prince
Fixes: [#5955](https://github.com/twisted/twisted/issues/5955)
